src/smart-components/OpenChannel/components/OpenChannelMessageList/useHandleOnScrollCallback.ts

```diff
@@ -0,0 +1,45 @@
+import React, { useCallback } from "react";
+import { SCROLL_BUFFER } from "../../../../utils/consts";
+
+export interface UseHandleOnScrollCallbackProps {
+  setShowScrollDownButton: React.Dispatch<React.SetStateAction<boolean>>;
+  hasMore: boolean;
+  onScroll(fn: () => void): void;
+  scrollRef: React.RefObject<HTMLDivElement>;
+}
+
+export function useHandleOnScrollCallback({
+  setShowScrollDownButton,
+  hasMore,
+  onScroll,
+  scrollRef,
+}: UseHandleOnScrollCallbackProps): (e: React.UIEvent<HTMLElement>) => void {
+  return useCallback((e) => {
+    const element = e.target as Element;
+    const {
+      scrollTop,
+      scrollHeight,
+      clientHeight,
+    } = element;
+    const scrollBottom = scrollHeight - scrollTop;
```

Comment:
Can you comment about why we do this calculation before merge? Also mention the issue :)